nvflare/app_common/pt/pt_model_reader_writer.py

```diff
@@ -29,11 +28,11 @@ def __init__(self):
     def initialize(self, trainer):
         pass
 
-    def extract_model(self, model_vars, fl_ctx: FLContext):
+    def extract_model(self, network, multi_processes, model_vars, fl_ctx: FLContext):
         # net = self.fitter.net
-        net = fl_ctx.get_prop(AppConstants.MODEL_NETWORK)
+        net = network
         # if self.fitter.multi_gpu:
-        if fl_ctx.get_prop(AppConstants.MULTI_GPU):
+        if multi_processes:
             net = net.module
         local_state_dict = net.state_dict()
```

Comment:
Should we add some defensive programming?  If net is not pytorch net, it does not always have state_dict().

Reply:
These components are in the app-common "pt" folder. They are going to only handle pytorch model network.

However, we can also add a check to test if the net is a pytorch net. Not sure if there's a simple way for this check. Otherwise, we need to wrap this in a try-catch block.